src/drivers/imx/esai.c

```diff
@@ -414,18 +414,32 @@ static int esai_get_handshake(struct dai *dai, int direction, int stream_id)
 	return EDMA_HANDSHAKE(irq, channel);
 }
 
-static int esai_get_fifo(struct dai *dai, int direction, int stream_id)
+static int esai_get_fifo(struct dai *dai, int direction, int id)
 {
 	switch (direction) {
 	case DAI_DIR_PLAYBACK:
 	case DAI_DIR_CAPTURE:
-		return dai_fifo(dai, direction); /* stream_id is unused */
+		return dai_fifo(dai, direction); /* id is unused */
 	default:
 		trace_esai_error("esai_get_fifo(): Invalid direction");
 		return -EINVAL;
 	}
 }
 
+static int esai_gen_dma_elem_array(struct dai *dai, int direction, int id,
+				   int periods, int period_bytes,
+				   uintptr_t buffer,
+				   struct dma_sg_array *sg_array)
+{
+	int ret = dma_sg_array_alloc(sg_array, 1, SOF_MEM_ZONE_RUNTIME);
+
+	if (ret)
```

Comment:
I have a preference for the `if (ret < 0)` style on error code (not blocking over this though)